### PR TITLE
VY-RES-453 -API for booked resources

### PR DESCRIPTION
**Pull request by lf-achyutpkl**
@lf-kailashraj  please review my code

### CODE REVIEW

File: src/main/java/com/lftechnology/vyaguta/resource/rs/ProjectRs.java [hunk 1]
```diff
@@ -1,10 +1,14 @@
 package com.lftechnology.vyaguta.resource.rs;
 
+import java.awt.image.renderable.ParameterBlock;
```
Comment (bhuwang) [inline src/main/java/com/lftechnology/vyaguta/resource/rs/ProjectRs.java]: MINOR Remove this unused import 'java.awt.image.renderable.ParameterBlock'. [rule](http://sonar.java.lftechnology.com:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/lftechnology/vyaguta/resource/rs/ProjectRs.java [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
 import java.util.List;
 import java.util.Map;
 import java.util.UUID;
 
 import javax.annotation.security.RolesAllowed;
+import javax.ejb.Local;
```
Comment (bhuwang) [inline src/main/java/com/lftechnology/vyaguta/resource/rs/ProjectRs.java]: MINOR Remove this unused import 'javax.ejb.Local'. [rule](http://sonar.java.lftechnology.com:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/lftechnology/vyaguta/resource/rs/ProjectRs.java [hunk 3]
```diff
@@ -103,4 +110,21 @@ public Response remove(@PathParam("id") UUID id) {
...
+    @Path("/resourceutilization")
+    @GET
+    @RolesAllowed({ "Employee", "Admin" })
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response resourceUtilization(@QueryParam("date") String dateStr) {
+        LocalDate date = LocalDate.now();
+        if (dateStr != null) {
+            try {
+                date = LocalDate.parse(dateStr, Constant.DATE_FORMAT_DB);
+            } catch (DateTimeParseException e) {
```
Comment (bhuwang) [inline src/main/java/com/lftechnology/vyaguta/resource/rs/ProjectRs.java]: CRITICAL Either log or rethrow this exception. [rule](http://sonar.java.lftechnology.com:9000/coding_rules#rule_key=squid%3AS1166)

File: src/main/java/com/lftechnology/vyaguta/resource/dao/impl/ContractMemberDaoImpl.java [hunk 4]
```diff
@@ -1,11 +1,15 @@
...
+import java.time.LocalDate;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.UUID;
 
 import javax.ejb.Stateless;
+import javax.persistence.Query;
+
+import org.apache.log4j.chainsaw.Main;
```
Comment (bhuwang) [inline src/main/java/com/lftechnology/vyaguta/resource/dao/impl/ContractMemberDaoImpl.java]: MINOR Remove this unused import 'org.apache.log4j.chainsaw.Main'. [rule](http://sonar.java.lftechnology.com:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by bhuwang** [comment]
SonarQube analysis reported 4 issues:
* CRITICAL 1 critical
* MINOR 3 minor

Watch the comments in this conversation to review them.

**Comment by bhuwang** [comment]
SonarQube analysis reported 4 issues:
* CRITICAL 1 critical
* MINOR 3 minor

Watch the comments in this conversation to review them.

File: src/main/java/com/lftechnology/vyaguta/resource/service/impl/EmployeeServiceImpl.java [hunk 5]
```diff
@@ -45,4 +45,18 @@
...
 
+    @Override
+    public List<Employee> fetchActiveEmployees() {
+        try {
+            String token = SecurityRequestContext.getAccessToken();
+            ResponseData<Employee> data = HttpHelper.get(EMPLOYEE_URL, token, new GenericType<ResponseData<Employee>>() {
+            });
+
+            return data.getData();
+        } catch (WebApplicationException e) {
```
Comment (anishmanandhar) [inline src/main/java/com/lftechnology/vyaguta/resource/service/impl/EmployeeServiceImpl.java]: WHy jax rs specific exception in service ??

Comment (lf-achyutpkl) [inline src/main/java/com/lftechnology/vyaguta/resource/service/impl/EmployeeServiceImpl.java]: this service communicates with another module and there is chances for facing this exception

Comment (anishmanandhar) [inline src/main/java/com/lftechnology/vyaguta/resource/service/impl/EmployeeServiceImpl.java]: Ok, this is alright .. i thought controller specific exception caught in the service. :+1:

**Pull request merged**